### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by martinyde**

### CODE REVIEW

File: src/CprSniffer.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        }
+    }
+
+    /**
+     * Check against valid date.
+     *
+     * @param $number
+     *   A string that could be a CPR number
+     *
+     * @return bool return True if it's considered a date
```
Comment (cableman) [inline src/CprSniffer.php:149]:
```suggestion
     * @return bool return 
     *   True if it's considered a date
```

File: src/CprSniffer.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+     */
+    private function dateChk($number)
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
    private function dateChk(string $number): bool
```

File: src/CprSniffer.php [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Check against valid date.
+     *
+     * @param $number
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
     * @param string $number
```

File: src/CprSniffer.php [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+            $array[$i] = substr($numString, $i, 1);
+        }
+
+        return $array;
+    }
+
+    /**
+     * Check against modulo 11.
+     *
+     * @param $array
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
     * @param array $array
```

File: src/CprSniffer.php [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+     *   An array of characters
+     * @param $number
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
     * @param string $number
```

File: src/CprSniffer.php [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+     *   A string that could be a CPR number
+     *
+     * @return bool True if the characters in union resemble a CPR number
```
Comment (cableman) [inline src/CprSniffer.php:97]:
```suggestion
     * @return bool 
     *   True if the characters in union resemble a CPR number
```

File: src/CprSniffer.php [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+     */
+    private function mod11Chk($array, $number)
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
    private function mod11Chk(array $array, string $number): bool
```

File: src/CprSniffer.php [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+    /**
+     * Create an array from a string.
+     *
+     * @param $numString
+     *   A string to create an array from
+     *
+     * @return array the created array of characters
+     */
+    private function stringSplit($numString)
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
    private function stringSplit(string $numString): array
```

File: src/CprSniffer.php [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+        }
+    }
+
+    /**
+     * Create an array from a string.
+     *
+     * @param $numString
+     *   A string to create an array from
+     *
+     * @return array the created array of characters
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
     * @return array<string>
     *   The created array of characters
```

File: src/CprSniffer.php [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+            if ($this->mod11Chk($arr, $number) && $this->dateChk($number)) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Create an array from a string.
```
Comment (cableman) [inline src/CprSniffer.php]: Why not use https://www.php.net/manual/en/function.str-split.php

> Converts a string to an array.

File: src/CprSniffer.php [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+     *
+     * @param $numString
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
     * @param string $numString
```

File: src/CprSniffer.php [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+        '010192',
+    ];
+
+    /**
+     * Check if string contains valid CPR.
+     *
+     * @param $string
+     *   The string to check
+     *
+     * @return bool true if string contains a number that could be considered cpr
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
     * @return bool 
     *   True if string contains a number that could be considered cpr
```

File: src/CprSniffer.php [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+        '010189',
+        '010190',
+        '010191',
+        '010192',
+    ];
+
+    /**
+     * Check if string contains valid CPR.
+     *
+     * @param $string
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
     * @param string $string
```

File: src/CprSniffer.php [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+     *   The string to check
+     *
+     * @return bool true if string contains a number that could be considered cpr
+     */
+    public function checkCpr($string)
```
Comment (cableman) [inline src/CprSniffer.php]:
```suggestion
    public function checkCpr(string $string): bool
```

File: src/CprSniffer.php [hunk 15]
```diff
@@ -0,0 +1,183 @@
+<?php
+
```
Comment (cableman) [inline src/CprSniffer.php:2]:
```suggestion

declare(strict_types=1);

```

**Review by cableman — CHANGES_REQUESTED** [review]
* Minor changes to enabled better type checking. 
* Maybe it should have been named `CPRValidator` not sure I understand the `sniffer` part.
* Maybe add php version 7.3 to composer (as php 7.2 and lower is EOL).

File: src/CprSniffer.php (hunk 10 above)
Comment (martinyde) [reply src/CprSniffer.php]: This was changed, and the function removed

**Comment by martinyde** [comment]
>     * Minor changes to enabled better type checking.
> 
>     * Maybe it should have been named `CPRValidator` not sure I understand the `sniffer` part.
> 
>     * Maybe add php version 7.3 to composer (as php 7.2 and lower is EOL).

- Added your suggested changes
- Added codesniffer config file.
- One lined variables and return value commenting in accordance with codesniffer.
- Added php 7.3 dependency
- About naming i grabbed the sniffer part from the name of the jira issue, thought it was your own naming. I'm not really invested in what the name should be so fell free to change.

**Pull request merged**